### ISSUE TITLE
[inductor][cpu] torch.isin returns False for a present inf when the test set contains nan

### ISSUE THREAD

**Issue by CaelusHu** [pytorch/pytorch#198484]
### 🐛 Describe the bug

Compiling `torch.isin` for CPU can return `False` for a `+inf` value that is present when the test
set also contains `nan`. Eager execution returns `True`.

Minimal reproducer:

```python
import torch

inf, nan = float("inf"), float("nan")
x = torch.tensor([inf])
test = torch.tensor([
    0.0, 1.0, -1.0, inf, -inf, nan,
    1e-38, 3.4e38, 0.5, -0.5, 0.0, 1.0,
])

fn = lambda a, b: torch.isin(a, b)

print(fn(x, test))
# tensor([True])

print(torch.compile(fn, fullgraph=True)(x, test))
# tensor([False])  <-- wrong
```

This reproduced in 2/2 clean CPU replay passes on the nightly build below. There is no warning,
graph break, or exception.

The result is correct with `backend="aot_eager"`, `backend="eager"`, `assume_unique=True`, a test
set without `nan`, or needles `-inf`, `3.4e38`, `1e-38`, and `1.0`. A length scan is wrong only
for test-set lengths `11`, `12`, `13`, `19`, and `20`.

The `isin_sorting` decomposition in `torch/_decomp/decompositions.py` sorts the test elements and
then does:

```python
idx = torch.searchsorted(sorted_test_elements, elements_flat)
cmp = sorted_test_elements[torch.where(idx < n, idx, 0)] == elements_flat
```

For the reproducer, the sorted set ends in `[..., inf, nan]`. On CPU, both eager and compiled
`searchsorted([inf])` return the out-of-range index `12`. The decomposition changes that index to
`0`, compares `-inf` with `+inf`, and returns `False`.

The full CUDA control is correct. In 2/2 passes, all five trigger shapes and seven controls match
eager, and every length from `4` through `20` is correct. Compiled CUDA `searchsorted([inf])`
returns `10`, the actual `inf` slot, while eager CUDA returns `12`; this explains why the compiled
CUDA decomposition avoids the CPU failure.

`isin_sorting` should handle non-finite values without assuming that `idx < n`. I have not tested
a source-level fix.

Issue #158738 discusses `searchsorted` with `nan`, but not `torch.isin` or `torch.compile`. Closed
PR #147519 changed the same decomposition for scalar and finite-value cases, not this `nan`/`inf`
case.

### Error logs

There is no exception. The failure is a silent wrong result:

```text
eager isin:         [True]
compiled CPU isin:  [False]
CPU searchsorted:   eager [12], compiled [12]
CUDA searchsorted:  eager [12], compiled [10]
```

Full control commands:

```bash
# CPU nightly
python repro.py --replays 2 --scan

# CUDA build
python repro.py --replays 2 --scan --cuda-full

# Both finish with: RESULT: 0 failing check(s).
```

### Versions

```text
CPU reproduction:
PyTorch version: 2.15.0.dev20260923+cpu
PyTorch git version: 54b9926ebf45a422247d8a72011dcbde9eb90f3d
Python version: 3.11.16
OS: Ubuntu 26.04 LTS, WSL2 (x86_64)
CPU: AMD Ryzen 5 9600X

CUDA control:
PyTorch version: 2.12.1+cu130
PyTorch git version: 7269437d655783a26cba32aa88195b741ff496aa
Python version: 3.12.14
CUDA version: 13.0
GPU: NVIDIA GeForce RTX 5060 Ti
NVIDIA driver: 581.29
```

cc @chauhang @penguinwu
